File: src/main/java/com/shipout/service/Userservice.java
package com.shipout.service;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.shipout.dao.UserMapper;
import com.shipout.entity.User;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.redis.core.StringRedisTemplate;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Objects;
import java.util.concurrent.TimeUnit;

@Service
public class Userservice {

    @Autowired
    private UserMapper userMapper;

    @Autowired
    private StringRedisTemplate stringRedisTemplate;

    @Autowired
    private ObjectMapper objectMapper;

    public User findUserById(int id) {
        try {
            String userString = stringRedisTemplate.opsForValue().get(id);
            if (Objects.isNull(userString)) {
                User user = userMapper.selectById(id);
                userString = objectMapper.writeValueAsString(user);
                stringRedisTemplate.opsForValue().set(String.valueOf(id), userString, 10, TimeUnit.MINUTES);
                return user;
            } else {
                return objectMapper.readValue(userString, User.class);
            }
        } catch (JsonProcessingException e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }


    }

    public List<User> getUserList() {
        return userMapper.selectList(null);
    }
}

File: src/main/java/com/shipout/service/UserService.kt
package com.shipout.service

import com.fasterxml.jackson.core.*
import com.fasterxml.jackson.databind.*
import com.shipout.dao.*
import com.shipout.entity.*
import org.springframework.beans.factory.annotation.*
import org.springframework.data.redis.core.*
import org.springframework.stereotype.*
import java.util.*
import java.util.concurrent.*

@Service
class UserService {
    @Autowired
    private var userMapper: UserMapper? = null

    @Autowired
    private lateinit var stringRedisTemplate: StringRedisTemplate

    @Autowired
    private lateinit var objectMapper: ObjectMapper
    fun findUserById(id: Int): User? {
        return try {
            var userString = stringRedisTemplate.opsForValue()[id]
            if (Objects.isNull(userString)) {
                val user = userMapper!!.selectById(id)
                userString = objectMapper!!.writeValueAsString(user)
                stringRedisTemplate.opsForValue().set(id.toString(), userString, 10, TimeUnit.MINUTES)
                user
            } else {
                objectMapper!!.readValue(userString, User::class.java)
            }
        } catch (e: JsonProcessingException) {
            e.printStackTrace()
            throw RuntimeException(e)
        }
    }

    fun getUserList(): List<User> = userMapper!!.selectList(null)
}
